src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+  const modifiedStates = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const type = actions[i].type;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(modifiedStates, actions[i].extraData);
+        statesArray.push({ ...modifiedStates });
```

Comment:
```suggestion
        statesArray.push(Object.assign(modifiedStates, actions[i].extraData));
```
or
```suggestion
        statesArray.push({ ...modifiedStates, ...actions[i].extraData });
```

Reply:
In these ways I add extraData only to a copy of an object that I push to an array. But according to the task I also need to change the object modifiedStates for the next iterations.